src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -182,13 +182,13 @@ private void OnSymbolActionCore(ISymbol symbol, Action<Diagnostic> reportDiagnos
                 if (symbol.Kind == SymbolKind.NamedType)
                 {
                     var namedType = (INamedTypeSymbol)symbol;
-                    if (namedType.InstanceConstructors.Length == 1 &&
-                        (namedType.TypeKind == TypeKind.Class || namedType.TypeKind == TypeKind.Struct))
+                    if ((namedType.InstanceConstructors.Length == 1 && namedType.TypeKind == TypeKind.Class)
```

Comment:
[cosmetic] I would move the `TypeKind` check first just so that when reading we have the 2 typekind check aligned.

Reply:
```suggestion
                    if ((namedType.TypeKind == TypeKind.Class && namedType.InstanceConstructors.Length == 1)
```